apps/heft/src/cli/actions/HeftActionBase.ts

```diff
@@ -180,12 +180,15 @@ export abstract class HeftActionBase extends CommandLineAction {
     }
   }
 
-  protected async parametersCallbackAsync(): Promise<void> {
+  protected async parametersCallbackExecuteAsync(): Promise<void> {
     const customParametersRecord: Record<string, CustomParameterType> = {};
     for (const [parameterName, getParameterValue] of this.customParametersMap.entries()) {
       customParametersRecord[parameterName] = getParameterValue();
     }
-    this.customParametersCallbacks.forEach(async (callback) => await callback(customParametersRecord));
+
+    await Promise.all(
+      [...this.customParametersCallbacks].map(async (callback) => await callback(customParametersRecord))
```

Comment:
See unresolved comment, this won't work quite right either.

Reply:
Thanks. This callback code is now replaced with something a bit better.